Fix Issue #3315: PixelShuffle_icnr icnr_init with weight_norm

pratX opened the pull request:
When using weight_norm (the current default), PixelShuffle_icnr needs to icnr_init weight_g and weight_v.
Issue https://github.com/fastai/fastai/issues/3315

pratX commented:
Work in Progress. Checking out nbdev and following the development process used for fastai.

pratX commented:
Updated the notebook with the fix and also the unit test. The unit test passes with weight_norm enabled (earlier weight_norm was explicitly disabled in the unit test, even though the description said that weight_norm is what worked best with super_resolution) which is the expected behavior.

pratX commented:
Updated PixelShuffle_icnr unit test to check for both cases, weight_norm enabled and disabled.

tcapelle commented:
Would you mind adding the spectral norm init also?

On Thu, Apr 15, 2021, 8:50 AM pratX ***@***.***> wrote:

> Updated PixelShuffle_icnr unit test to check for both cases, weight_norm
> enabled and disabled.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/fastai/fastai/pull/3322#issuecomment-820164227>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEMWOAK4ANPJKXU26YDGKX3TI2EDDANCNFSM42542OBA>
> .
>

pratX commented:
> Would you mind adding the spectral norm init also?
> […](#)
> On Thu, Apr 15, 2021, 8:50 AM pratX ***@***.***> wrote: Updated PixelShuffle_icnr unit test to check for both cases, weight_norm enabled and disabled. — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub <[#3322 (comment)](https://github.com/fastai/fastai/pull/3322#issuecomment-820164227)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AEMWOAK4ANPJKXU26YDGKX3TI2EDDANCNFSM42542OBA> .

Added unit test for PixelShuffle_icnr with sepctral_norm.

tcapelle commented:
graet!

jph00 commented:
Many thanks!
